patches/server/0925-Paper-Plugins.patch

```diff
@@ -0,0 +1,1274 @@
++                this.status = ProviderStatus.ERRORED;
++                SneakyThrow.sneaky(ex);
++            }
++
++            throw new IllegalStateException(); // Shouldn't reach this point
++        }
++
++        @Override
++        public PluginConfiguration getConfiguration() {
++            return PaperPluginProvider.this.description;
++        }
++
++        @Override
++        public Logger getLogger() {
++            return PaperPluginProvider.this.logger;
++        }
++
++        @Override
++        public ProviderStatus getLastProvidedStatus() {
++            return this.status;
++        }
++
++        @Override
++        public void setStatus(ProviderStatus status) {
++            this.status = status;
++        }
++
++        public boolean shouldSkipCreation() {
++            return this.initializerProvider.status == ProviderStatus.ERRORED;
++        }
++    }
++
++}
+diff --git a/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProvider.java b/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProvider.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..77d81c3aa4fcfafa33bb7ee796c7558834815c3b
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/plugin/provider/type/spigot/SpigotPluginProvider.java
+@@ -0,0 +1,118 @@
++package io.papermc.paper.plugin.provider.type.spigot;
++
++import com.destroystokyo.paper.util.SneakyThrow;
++import io.papermc.paper.plugin.provider.ProviderStatus;
++import io.papermc.paper.plugin.provider.ProviderStatusHolder;
++import io.papermc.paper.plugin.provider.type.JavaPluginProvider;
++import org.bukkit.Bukkit;
++import org.bukkit.plugin.InvalidDescriptionException;
++import org.bukkit.plugin.InvalidPluginException;
++import org.bukkit.plugin.PluginDescriptionFile;
++import org.bukkit.plugin.SimplePluginManager;
++import org.bukkit.plugin.java.JavaPlugin;
++import org.jetbrains.annotations.NotNull;
++import org.slf4j.Logger;
++import org.slf4j.LoggerFactory;
++import org.yaml.snakeyaml.error.YAMLException;
++
++import java.io.IOException;
++import java.io.InputStream;
++import java.nio.file.Path;
++import java.util.jar.JarEntry;
++import java.util.jar.JarFile;
++
++public class SpigotPluginProvider extends JavaPluginProvider implements ProviderStatusHolder {
++
++    public static SpigotPluginProvider of(JarFile file, JarEntry config, Path pluginPath) throws Exception {
++        PluginDescriptionFile descriptionFile;
++        try (InputStream inputStream = file.getInputStream(config)) {
++            // throw new InvalidDescriptionException(new FileNotFoundException("Jar does not contain plugin.yml")); TODO: This will no longer happen. Fix this?
++            descriptionFile = new PluginDescriptionFile(inputStream);
++        } catch (IOException | YAMLException ex) {
++            throw new InvalidDescriptionException(ex);
++        }
++
++        // TODO ORIGINAL CODE: Should I keep logging it or should this be an exception?
```

Comment:
The plugin will not be loaded at this point maybe a throw is better to avoid sneaky issue when we think this plugins work but then not really